source/beamline/BioXAS/BioXAS32ElementGeDetector.cpp

```diff
@@ -31,3 +34,42 @@ bool BioXAS32ElementGeDetector::setAcquisitionTime(double seconds)
 
 	return true;
 }
+
+AMAction3* BioXAS32ElementGeDetector::createEraseAction()
+{
+	AMAction3 *result = 0;
+
+	// Build the erase action.
+
+	AMListAction3* eraseAction = new AMListAction3(new AMListActionInfo3("Erase detector data", "Erase detector data"), AMListAction3::Sequential);
+
+	// Begin by manually toggling the erase control.
+
+	eraseAction->addSubAction(AMActionSupport::buildControlMoveAction(eraseControl_, 1.0));
+	eraseAction->addSubAction(AMActionSupport::buildControlMoveAction(eraseControl_, 0.0));
+
+	// Iterate through the SCA controls we care about, making sure they all get set to 0.
+
+	bool controlsValid = true;
+
+	for (int i = 0, count = elements(); i < count && controlsValid; i++) {
+		if (isElementEnabled(i)) {
+			AMControl *icrControl = icrControlAt(i);
+
+			if (icrControl)
+				eraseAction->addSubAction(AMActionSupport::buildControlWaitAction(icrControl, 0.0));
+			else
+				controlsValid = false;
+		}
+	}
+
+	// If there was no problem with the controls, then the result is the generated erase action.
+	// Otherwise, generate error to notify user.
+
+	if (controlsValid)
+		result = eraseAction;
+	else
+		AMErrorMon::alert(this, BIOXAS32ELEMENTGEDETECTOR_ERASE_FAILED, QString("The detector '%1' attempted to erase data and failed. Please report to beamline staff.").arg(name()));
```

Comment:
Two questions here:
1) do we want to know which element (`icrControl`) is invalid in the alert message? I think it might be a useful information to tackle the issue?
2) do we want NOT to do any  data cleanup action for the valid controls if there is one (or some) invalid control(s)?